manager/pkg/cluster/cluster.go

```diff
@@ -0,0 +1,141 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/gke-rolling-updates-demo/manager/pkg/utils"
+
+	container "cloud.google.com/go/container/apiv1"
+	containerpb "google.golang.org/genproto/googleapis/container/v1"
+)
+
+type ManagedCluster struct {
+	Client      *container.ClusterManagerClient
+	Cluster     *containerpb.Cluster
+	Project     string
+	Location    string
+	ClusterName string
+	NodeCount   int32
+}
+
+func NewManagedCluster(client *container.ClusterManagerClient, project string, location string, clusterName string, nodeCount int32) *ManagedCluster {
+	return &ManagedCluster{
+		Client:      client,
+		Project:     project,
+		Location:    location,
+		ClusterName: clusterName,
+		NodeCount:   nodeCount,
+	}
+}
+
+func (c *ManagedCluster) Create(ctx context.Context) error {
+	getReq := &containerpb.GetClusterRequest{
+		Name: fmt.Sprintf("projects/%s/locations/%s/clusters/%s", c.Project, c.Location, c.ClusterName),
+	}
+	resp, err := c.Client.GetCluster(ctx, getReq)
+	if resp != nil {
+		c.Cluster = resp
+		return nil
+	}
+	req := &containerpb.CreateClusterRequest{
```

Comment:
what does req return?  Do we need to check the value?

Reply:
`req` itself doesn't return anything. It's just a data structure used to passed the desired values to the function that executes the API call.